### ISSUE TITLE
PEP 561 Comptability

### ISSUE THREAD

**Issue by mina-asham** [mattjw/sparkql#117]
TL;DR; PEP 561 is stricter than current typing, I suggest replacing `pytype` with `mypy` and adding `py.typed` file in output package, happy to submit a PR if we agree.

Type checking support has been added as part of #27, but I noticed that the package doesn't distribute `py.typed` marker.

I thought adding `py.typed` would be enough, but looks like the package is not 100% [PEP 561](https://peps.python.org/pep-0561/) compatible, checking the previous work it seems like `pytype` was integrated but it seems it still doesn't support [PEP 561](https://peps.python.org/pep-0561/) properly: https://github.com/google/pytype/issues/151 (that might be because pytype was designed to be more lenient and rely on inference more?)

mypy on the other hand has a strict mode and should be able to guide us through being 100% PEP 561 compatible.

I am more than happy submitting a PR switching to mypy and adding py.typed marker file if we are okay with this here? Let me know what you think

**Comment by mattjw** [comment]
@mina-asham I'd love sparkql to migrate to mypy and PEP 561 compliance. Since originally setting this project up I've moved to mypy in my own personal and professional projects. It'd be nice for sparkql to catch up, too.

If you have time to raise a PR that would be wonderful 🙏.

**Comment by mina-asham** [comment]
Awesome, thanks for the prompt response and for the super useful library, here is my PR for supporting PEP 561: https://github.com/mattjw/sparkql/pull/119
